### PR TITLE
Make strategies configurable

### PR DESCRIPTION
**Pull request by kirs**
Some apps don't want to use `CSSLoaders::NetworkLoader` for security reasons. Now they have to mutate private `Premailer::Rails::CSSHelper::STRATEGIES` constant to skip the network loader.

This PR introduces `strategies` config key that allows to specify strategies that an app wants to use:

```ruby
Premailer::Rails.config.merge!(strategies: [:asset_pipeline, :file_system])
```

### CODE REVIEW

**Comment by kirs** [comment]
@artygus @fphilipe let me know what you think about this config option.

File: lib/premailer/rails/css_helper.rb [hunk 1]
```diff
@@ -36,14 +34,33 @@ def css_urls_in_doc(doc)
         end
       end
 
+      def load_css_with_cache(url)
+        self.cache[url] ||= load_css(url)
+      end
+
+      def cache_enabled?
+        !::Rails.env.development?
```
Comment (fphilipe) [inline lib/premailer/rails/css_helper.rb]: Let's check for `defined?(::Rails)` before using it as described in https://github.com/fphilipe/premailer-rails/pull/201/files#r149874772.

File: lib/premailer/rails/css_helper.rb [hunk 2]
```diff
@@ -36,14 +34,35 @@ def css_urls_in_doc(doc)
         end
       end
 
+      def load_css_with_cache(url)
+        self.cache[url] ||= load_css(url)
+      end
+
+      def cache_enabled?
+        defined?(::Rails) && ::Rails.env.production?
```
Comment (kirs) [inline lib/premailer/rails/css_helper.rb]: @fphilipe I think it makes sense to only use cache in production (before it was !development, which affected test). WDYT?

**Comment by kirs** [comment]
Thanks for feedback! I rebased it and addressed all comments 👌

**Comment by fphilipe** [comment]
Landed in 73ff689. Thanks! Would you mind taking it for a spin before I cut a release, @kirs?

**Comment by kirs** [comment]
It has been running in production with no issues for a while 👌

**Comment by fphilipe** [comment]
@kirs thanks for your contribution. I've released version 1.10.0 just now.

**Pull request closed without merging**